game/game/models/world.py

```diff
@@ -77,6 +77,14 @@ def get_character_by_id(self, char_id: ctype.int32) -> typing.Union[None, "Chara
     def find_object_by_id(self, object_id: ctype.int32) -> typing.Union[None, L2Object]:
         return self._objects.get(object_id)
 
+    def get_session_by_character(self, character: "Character") -> "GameSession":
+        return self._characters.get(character)
```

Comment:
Not sure that this will work if you pass two different instances of the same Character